File: BallClockMins.rb
# BallClockMins
# By Joseph Sturzenegger

require_relative 'BallClock'
require 'json'

class BallClockMins < BallClock
	# creates a ball clock with the number of balls and a minute count to know the time elapsed
	def initialize(ball_count, minute_count)
		super(ball_count)
		@minute_count = minute_count
	end

	# runs the clock for the number of specified minutes and prints the JSON object of the clock state
	def run_clock
		for i in (1..@minute_count)
			add_minute(@main.shift)
		end
		json_obj_hash = {:Min => @mins, :FiveMin => @five_mins, :Hour => @hours, :Main => @main}
		puts JSON.generate(json_obj_hash)
	end

end

File: BallClock/BallClock/BallClockMins.swift
//
//  BallClockMins.swift
//  BallClock
//
//  Created by Joe Sturzenegger on 5/5/15.
//  Copyright (c) 2015 Joe Sturzenegger. All rights reserved.
//

import Foundation

class BallClockMins: BallClock
{
    var minuteCount: Int
    init(ballCount: Int, minuteCount: Int)
    {
        self.minuteCount = minuteCount
        super.init(ballCount: ballCount)
    }
    
    override func runClock()
    {
        for i in (1...minuteCount)
        {
            addMinute(mainQueue.removeAtIndex(0))
        }
        var jsonObjectHash = getJSON()
        println(jsonObjectHash)
    }
    func getJSON()->String
    {
        
        var min: String = "\"Min\":\(mins),"
        var five: String = "\"FiveMin\":\(fiveMins),"
        var hr: String = "\"Hour\":\(hours),"
        var main: String = "\"Main\":\(mainQueue)"
        
        return "{" + min + five + hr + main + "}"
    }
}